src/playground/components/RuleItem.js

```diff
@@ -4,10 +4,8 @@ export default function RuleItem({ children }) {
     const ref = useRef();
 
     useEffect(() => {
-        if (ref.current) {
```

Comment:
I suppose `current`  is defined only when we set the `ref` to a DOM element.

Reply:
No No when ever we create a ref its value is usually { current: `initialrefvalue` }

Reply:
What if we don't have an initial ref?

Reply:
usually return is first executed in react and then useEffect so we already have a reference in return to a list item. Then while executing useEffect we have the reference to the dom element 🙈 .

Reply:
- `ref` could be `undefined` to begin with, and then set later.
- `ref` could change to a different element.
- `ref` could change from an element to `undefined`.

Reply:
When set to a element how can it become undefined 😅 . May be check demo? It should work and wouldn't break?

Reply:
I mean yes, this will work here. My point was that it's possible for it to be `undefined` if we try to add more code here in the future.
For now, I'm ok with this change.

Reply:
Yep sure